master/internal/command/command_service.go

```diff
@@ -131,6 +131,59 @@ func (cs *CommandService) LaunchGenericCommand(
 	return cmd, nil
 }
 
+// LaunchNotebookCommand creates notebook commands and persists them to the database.
+func (cs *CommandService) LaunchNotebookCommand(
```

Comment:
I'm a little worried about this logic diverging from the generic one ```LaunchGenericCommand``` 

can we instead have ```LaunchGenericCommand``` check task type and only create a notebook session if task type is notebook?

Reply:
yeah, i didn't want this either, but it's hard to isolate, since the notebook launch takes in a session parameter and also sprinkles in additional calls throughout the method.